packages/components/src/modal/index.tsx

```diff
@@ -179,6 +188,9 @@ function UnforwardedModal(
 				overlayClassName
 			) }
 			onKeyDown={ handleEscapeKeyDown }
+			onPointerDown={
```

Comment:
Can we add a TODO comment here? Something to the effect of "Reconcile with `useFocusOutside`" and "Ideally this should be an onClick".

Reply:
I gave it my best and hopefully it captures what you had in mind. At least regarding the first part. I left out the bit about preferring `onClick`. That could provide some nuance to the interaction but I think handling `pointerdown` will still be necessary. That’s because if the default isn't prevented from 'pointerdown' focus will move and be left unreturned by `useFocusReturn`.

Reply:
It does 👍 I'm mostly concerned that we're treating a mere `pointerdown` as a `click` intent, which isn't technically true. For example I sometimes `mousedown` on a button but change my mind and drag away so it doesn't go through as a `click`.

But I guess we could even resolve that right now, if we just `event.preventDefault()` on the `pointerdown`, and wait until the `click` event to actually call `onRequestClose()`.

Reply:
Yep, that's just what I meant about nuance in the interaction. I do the same sometimes. To keep current behavior and this PR is focused I'd left it as is.

Since you agree it should be resolved I went ahead and tried it. It's not quite as simple as we likely both thought. It's also opportunity try a bit of further nuance as suggested by Aki https://github.com/WordPress/gutenberg/pull/51602#issuecomment-1600392962. So I've got a branch for it and will make it a separate PR.

Reply:
…and that PR #52994